src/app/community-list-page/community-list-datasource.ts

```diff
@@ -21,10 +22,10 @@ export class CommunityListDatasource implements DataSource<FlatNode> {
     return this.communityList$.asObservable();
   }
 
-  loadCommunities(expandedNodes: FlatNode[]) {
+  loadCommunities(findOptions: FindListOptions, expandedNodes: FlatNode[]) {
```

Comment:
This method can take quite long to finish, and blocks the UI while it's loading. So I'd use zones to keep it from blocking while it's loading, and only update again when it's finished. e.g.:

```suggestion
  loadCommunities(findOptions: FindListOptions, expandedNodes: FlatNode[]) {
    this.loading$.next(true);
    this.zone.runOutsideAngular(() => {
      this.communityListService.loadCommunities(findOptions, expandedNodes).pipe(
        take(1),
        finalize(() => this.loading$.next(false)),
      ).subscribe((flatNodes: FlatNode[]) => {
        this.zone.run(() => this.communityList$.next(flatNodes));
      });
    })
  }
```
here, and everywhere else loadCommunities is used.

this.zone can be injected as a service in your constructor: 
```
private zone: NgZone
```